src/main/scala/io/iohk/ethereum/domain/Checkpoint.scala

```diff
@@ -0,0 +1,31 @@
+package io.iohk.ethereum.domain
+
+import io.iohk.ethereum.crypto.ECDSASignature
+import io.iohk.ethereum.rlp._
+
+case class Checkpoint(signatures: Seq[ECDSASignature])
+
+object Checkpoint {
+
+  import io.iohk.ethereum.crypto.ECDSASignatureImplicits._
+
+  implicit val checkpointRLPEncoder: RLPEncoder[Checkpoint] = _.toRLPEncodable
+
+  implicit val checkpointRLPDecoder: RLPDecoder[Checkpoint] = {
+    case RLPList(signatures: RLPList) =>
+      Checkpoint(
+        signatures.items.map(ecdsaSignatureDec.decode)
+      )
+    case _ => throw new RuntimeException("Cannot decode Checkpoint")
+  }
+
+  implicit class CheckpointEnc(checkpoint: Checkpoint) extends RLPSerializable {
```

Comment:
Do we need this class? It would be more concise to express it as a SAM type (same as the decoder).